django/core/cache/backends/redis.py

```diff
@@ -0,0 +1,239 @@
+"""Redis cache backend."""
+
+import pickle
+import random
+import re
+import urllib
+
+import redis
+
+from django.core.cache.backends.base import DEFAULT_TIMEOUT, BaseCache
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.functional import cached_property
+from django.utils.module_loading import import_string
+
+
+class BaseCacheSerializer:
+    def dumps(self):
+        raise NotImplementedError
+
+    def loads(self):
+        raise NotImplementedError
+
+
+class PickleCacheSerializer(BaseCacheSerializer):
```

Comment:
I don't think there is any point in creating `BaseCacheSerializer` and we can rename `PickleCacheSerializer`:
```suggestion
class PickleSerializer:
```

Reply:
I did this so that people can extent the `BaseCacheSerializer` to implement custom serializers. For example, using a msgpack serializer or a yaml serializer as most third party plugin provide this customization. 
Should I remove the `BaseCacheSerializer` and leave serializer customizations for the third-party plugins?

Reply:
I think this can be removed for now. Ultimately we should allow passing the serializer to the client class and just document that it requires `.dumps()` and `.loads()` methods.

Again, it just feels a little bit too much at this stage, and it's not hard to add that interface back in if we need to later. I'm just aiming for simplicity to help us with determining what options will be exposed by the client constructor.